### PR TITLE
Schema migration

### PR DESCRIPTION
**Pull request by massadm**
| ?            |  !
|---           | ---
| Type         | feature
| BC Break     | no
| Fixed issues | #1186 

#### Summary

Schema migration

### CODE REVIEW

**Comment by massadm** [comment]
Please review and let me know if changes can be moved to the `structure.sql`.

File: utils/db-schema-update/0.71.0-unreleased.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Invite Links
+CREATE TABLE IF NOT EXISTS `chat_invite_link` (
+    `id` BIGINT UNSIGNED COMMENT 'Unique identifier for this entry'
+        PRIMARY KEY,
+    `invite_link` VARCHAR(2083) CHARACTER SET 'ascii' COLLATE 'ascii_general_ci' NOT NULL COMMENT 'The invite link. If the link was created by another chat administrator, then the second part of the link will be replaced with “…”',
```
Comment (noplanman) [inline utils/db-schema-update/0.71.0-unreleased.sql:5]: We should be safe with much less here, I really don't think an invite link will have that many characters.

Also, why the collation change for this field?

Comment (massadm) [reply utils/db-schema-update/0.71.0-unreleased.sql:5]: > We should be safe with much less here, I really don't think an invite link will have that many characters.

_VARCHAR values are stored as a one-byte or two-byte length prefix plus data. A VARCHAR column uses one length byte if values require no more than 255 bytes, two length bytes if values may require more than 255 bytes._

So, `VARCHAR(255)` or `VARCHAR(2083)` is real options. I leave it up to you.

> Also, why the collation change for this field?

I am assuming here that the URI will be stored in transport form which is 7-bit coded character set, and in addition, non-ASCII internationalized domain names intended for resolution via the DNS will be IDNA encoded as well.

Comment (noplanman) [reply utils/db-schema-update/0.71.0-unreleased.sql:5]: Ok, thanks for the explanation. I think 255 and staying with `utf8mb4` is more than enough.

What advantage do you see in using `ascii` vs. the `utf8mb4`?

Comment (massadm) [reply utils/db-schema-update/0.71.0-unreleased.sql:5]: > What advantage do you see in using ascii vs. the utf8mb4?

UUIDs or URIs are clearly ASCII so CHARACTER SET 'ascii' is suitable domain for storing data like this.

**Comment by noplanman** [comment]
@massadm Thanks for your contribution, you can go ahead and add the new tables and fields to the `structure.sql` file 🙏 

Remember that the tables also need to be populated in `DB.php`, but I'm happy to do that part, which can be a bit confusing with the codebase as it is now 😁

**Pull request merged**